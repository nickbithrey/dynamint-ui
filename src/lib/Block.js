import React from 'react';
import { Stack } from 'office-ui-fabric-react/lib/Stack';

const Block = ({children, ...remainingProps}) => (
	<Stack.Item {...remainingProps}>
		{children}
	</Stack.Item>
);

export default Block;