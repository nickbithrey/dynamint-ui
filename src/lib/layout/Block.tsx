import * as React from 'react';
import { Stack, IStackItemStyles, IStackItemTokens } from 'office-ui-fabric-react/lib/Stack';

export interface IBlock {
    children: React.ReactNode;
    tokens?: IStackItemTokens;
    styles?: IStackItemStyles;
    align?: 'auto' | 'stretch' | 'baseline' | 'start' | 'center' | 'end';
    order?: number;
    disableShrink?: boolean;
    shrink?: number | boolean | 'initial' | 'inherit' | 'unset';
    verticalFill?: boolean;
}

export const Block = ( {
    children,
    shrink = 'initial',
    align = 'auto',
    disableShrink = false,
    verticalFill = true,
    ...itemProps
}: IBlock ) => {
    return (
        <Stack.Item 
            shrink={shrink} 
            align={align} 
            disableShrink={disableShrink} 
            verticalFill={verticalFill} 
            {...itemProps}>
            {children}
        </Stack.Item>
    );
}