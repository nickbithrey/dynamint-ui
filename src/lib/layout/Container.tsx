import * as React from 'react';
import { Stack, IStackStyles, IStackTokens } from 'office-ui-fabric-react/lib/Stack';

export interface IContainer {
    children?: React.ReactNode;
    tokens?: IStackTokens,
    styles?: IStackStyles
}

export const Container = ({children, tokens, styles}: IContainer) => {
    if (!children) {
        return null;
    }
    return (
        <Stack tokens={tokens} styles={styles}>
            {children}
        </Stack>
    );
}