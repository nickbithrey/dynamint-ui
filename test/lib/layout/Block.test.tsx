import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Block } from 'lib/layout/Block';

const expectedTag = 'StackItem';

describe('<Block />', () => {
    
    const props = {
        tokens: {
            margin: 20,
            padding: 10
        }
    }

    it('renders valid tag and matches snapshot with default values', () => {
        const wrapper: ShallowWrapper = shallow(
            <Block {...props}>
                <p>test</p>
            </Block>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it('will use default properties when not supplied', () => {
        const wrapper: ShallowWrapper = shallow(
            <Block {...props}>
                <p>test</p>
            </Block>
        );
        expect(wrapper.find(expectedTag).prop('shrink')).toEqual('initial');
        expect(wrapper.find(expectedTag).prop('align')).toEqual('auto');
        expect(wrapper.find(expectedTag).prop('disableShrink')).toBeFalsy();
        expect(wrapper.find(expectedTag).prop('verticalFill')).toBeTruthy();
    });
    
    it('will use overridden properties when supplied', () => {
        const wrapper: ShallowWrapper = shallow(
            <Block 
                shrink="inherit" 
                align="stretch" 
                disableShrink={true} 
                verticalFill={false} 
                {...props}>
                <p>test</p>
            </Block>
        );
        expect(wrapper.find(expectedTag).prop('shrink')).toEqual('inherit');
        expect(wrapper.find(expectedTag).prop('align')).toEqual('stretch');
        expect(wrapper.find(expectedTag).prop('disableShrink')).toBeTruthy();
        expect(wrapper.find(expectedTag).prop('verticalFill')).toBeFalsy();
    });
    
    it('will pass props to main item', () => {
        const wrapper: ShallowWrapper = shallow(
            <Block {...props}>
                <p>test</p>
            </Block>
        );
        expect(wrapper.find(expectedTag).prop('tokens')).toBe(props.tokens);
    });
    
    it('has default props set', () => {
        const wrapper: ShallowWrapper = shallow(
            <Block {...props}>
                <p>test</p>
            </Block>
        );
        const tag = wrapper.find(expectedTag);
        expect(tag.prop('align')).toBe('auto');
        expect(tag.prop('disableShrink')).toBeFalsy();
        expect(tag.prop('shrink')).toBe('initial');
        expect(tag.prop('verticalFill')).toBeTruthy();
    })
    
});