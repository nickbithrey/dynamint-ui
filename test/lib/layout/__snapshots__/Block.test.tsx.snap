// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> renders valid tag and matches snapshot with default values 1`] = `
<StackItem
  align="auto"
  disableShrink={false}
  shrink="initial"
  tokens={
    Object {
      "margin": 20,
      "padding": 10,
    }
  }
  verticalFill={true}
>
  <p>
    test
  </p>
</StackItem>
`;
